src/Api/Client.php

```diff
@@ -10,75 +10,56 @@
 namespace Flarum\Api;
 
 use Exception;
-use Flarum\Foundation\ErrorHandling\JsonApiFormatter;
-use Flarum\Foundation\ErrorHandling\Registry;
 use Flarum\Http\RequestUtil;
 use Flarum\User\User;
 use Illuminate\Contracts\Container\Container;
-use InvalidArgumentException;
 use Laminas\Diactoros\ServerRequestFactory;
-use Psr\Http\Message\ResponseInterface;
-use Psr\Http\Server\RequestHandlerInterface;
-use Throwable;
+use Laminas\Stratigility\MiddlewarePipeInterface;
+use Psr\Http\Message\ResponseInterface as Response;
+use Psr\Http\Message\ServerRequestInterface as Request;
 
 class Client
 {
     /**
-     * @var Container
+     * @var MiddlewarePipeInterface
      */
-    protected $container;
-
-    /**
-     * @var Registry
-     */
-    protected $registry;
+    protected $pipe;
 
     /**
      * @param Container $container
-     * @param Registry $registry
      */
-    public function __construct(Container $container, Registry $registry)
+    public function __construct(MiddlewarePipeInterface $pipe)
     {
-        $this->container = $container;
-        $this->registry = $registry;
+        $this->pipe = $pipe;
     }
 
     /**
      * Execute the given API action class, pass the input and return its response.
      *
-     * @param string|RequestHandlerInterface $controller
+     * @param string $routeName
+     * @param Request|null $parent
      * @param User|null $actor
      * @param array $queryParams
      * @param array $body
      * @return ResponseInterface
```

Comment:
The `use` alias hasn't been updated here.

Is it really necessary to alias request and response to shorter names though?